agent/consul/state/intention.go

```diff
@@ -49,10 +49,14 @@ func intentionsTableSchema() *memdb.TableSchema {
 				Name:         "source",
 				AllowMissing: true,
 				// This index is not unique since we need uniqueness across the whole
-				// 4-tuple.
+				// 5-tuple.
 				Unique: false,
 				Indexer: &memdb.CompoundIndex{
 					Indexes: []memdb.Indexer{
+						&memdb.StringFieldIndex{
```

Comment:
This is needed for intention matching to support matching an external URI against a trust domain intention, e.g. `spiffe://example.com/path` needs to match against a trust domain intention for `spiffe://example.com`. To do this, we need to be able to search on intention type.